include/deal.II/lac/block_linear_operator.h

```diff
@@ -662,6 +662,208 @@ block_triangular_inverse(const BlockMatrix &block_matrix,
   return result * inverse_diagonal;
 }
 
+/**
+ * @relates LinearOperator
+ *
+ * This function implement a forward substitution argument to invert a lower
+ * block triangular matrix.
+ * It takes as argement a block matrix @p block_matrix and an array of LinearOperators
+ * @p inverse_diagonal representing inverses of digonal blocks of @p block_matrix.
+ *
+ * Let us assume we have a linear system where each coefficient of the system is a
+ * matrix:
+ * A00 x0 + ...                   = y0
+ * A01 x0 + A11 x1 + ...          = y1
+ * ...        ...
+ * A0n x0 + A1n x1 + ... + Ann xn = yn
+ *
+ * First of all, x0 = A00^-1 y0
+ * Then, we can use x0 to recover x1:
+ *    x1 = A11^-1 ( y1 - A01 x0 )
+ * and therefore:
+ *    xn = Ann^-1 ( yn - A0n x0 - ... - A(n-1)n x(n-1) )
+ *
+ * Caveat: Tvmult and Tvmult_add have not been implemented, yet. This may lead to mistakes.
+ * @ingroup LAOperators
+ */
+
+template <typename BlockMatrix,
+          unsigned int n  = BlockMatrix::n ,
+          typename Range  = typename BlockMatrix::BlockType>
+LinearOperator<Range, Range>
+block_forward_substitution(const BlockMatrix &block_matrix,
+                           const std::array<LinearOperator<typename Range::BlockType, typename Range::BlockType>, n> &inverse_diagonal)
```

Comment:
Does this compile? In other words, does BlockMatrix always have a local static const variable `n` and a local typedef `BlockType`? If so, then yes this works.

I had in mind to write it as
```
template <typename Range, typename Domain, typename BlockMatrix, int n>
```
The last two arguments can be deduced by the compiler, so you don't have to explicitly specify them in the call. The first two cannot be deduced, so you have to explicitly specify them.